### DISCUSSION TITLE
Creating a plugin using TypeScript

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [tailwindlabs/tailwindcss#2502] [Help] [answered]
Does anyone have experience using TypeScript to create a plugin for Tailwind?

As far as I can tell, you have to compile the TS into JS, then use the JS version. Any way to use TS directly?

**Comment by RobinMalfait** [comment, marked as answer]
As far as I know, you will have to compile to JS because of the following reasons:

1. Tailwind CSS itself is not written in TypeScript
2. Tailwind CSS uses the commonjs module system
3. Because of this ^ it can still work in older version like node 8

So for now you will need to transpile it yourself. In the future we might be able to accept tailwind configs that are written in TypeScript directly, but then *we* need to transpile it. Either transpile it beforehand or at runtime using something like `ts-node`.

**Reply by Nick-Mazuk** [reply]
Thanks Robin!

**Comment by JNavith** [comment]
You may also want to see previous discussion https://github.com/tailwindlabs/tailwindcss/discussions/2310#discussioncomment-61674
